src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,228 @@
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
+                ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        this.client = client;
+    }
+
+    /**
+     * Load data for a given CType
+     * @param config CType whose data is to be loaded in-memory
+     * @return configuration loaded with given CType data
+     */
+    protected static final SecurityDynamicConfiguration<?> load(final CType config, boolean logComplianceEvent) {
+        SecurityDynamicConfiguration<?> loaded = configurationRepository.getConfigurationsFromIndex(Collections.singleton(config), logComplianceEvent).get(config).deepClone();
+        return DynamicConfigFactory.addStatics(loaded);
+    }
+
+    protected void saveAndUpdateConfiguration(final Client client, final CType cType,
+                                              final SecurityDynamicConfiguration<?> configuration) {
+        final IndexRequest ir = new IndexRequest(this.securityIndex);
+
+        final String id = cType.toLCString();
+
+        configuration.removeStatic();
+
+        try {
+            client.index(ir.id(id)
+                            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                            .setIfSeqNo(configuration.getSeqNo())
+                            .setIfPrimaryTerm(configuration.getPrimaryTerm())
+                            .source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)));
+        } catch (IOException e) {
+            throw ExceptionsHelper.convertToOpenSearchException(e);
+        }
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param accountDetailsAsString A string JSON of different account configurations.
+     * @return
+     * @throws UserServiceException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(String accountDetailsAsString) throws IOException {
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode accountDetails = mapper.readTree(accountDetailsAsString);
+        final ObjectNode contentAsNode = (ObjectNode) accountDetails;
+        final SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getConfigName(), false);
+
+        String accountName = securityJsonNode.get("name").asString();
+
+        if (accountName == null || accountName.length() == 0) { // Fail if field is present but empty
+            throw new UserServiceException(NO_ACCOUNT_NAME_MESSAGE);
+        }
+
+        final Map<String, String> accountAttributes = new HashMap<>();
+
+        if (!securityJsonNode.get("service").isNull() && securityJsonNode.get("service").asString() == "true") { // If this is a service account
+
+            accountAttributes.put("service", "true"); // This attribute signifies that the account is a service account
```

Comment:
I remember we had a conversation about the activation/deactivation about service account before. I was wondering can this flag be changed dynamically for those activation and deactivation?

Reply:
It will be able to be changed with an API once we implement that. Right now, it will just be part of the configuration file.